src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java

```diff
@@ -738,7 +741,12 @@ public void valueChanged(ListSelectionEvent evt) {
                         setDirectorySelected(true);
                         setDirectory(file);
                         if (usesSingleFilePane) {
-                            chooser.setSelectedFile(null);
+                            Path path = Paths.get(file.getPath());
+                            if (Files.isSymbolicLink(path)) {
+                                chooser.setSelectedFile(file);
+                            } else {
+                                chooser.setSelectedFile(null);
+                            }
```

Comment:
What about the case where multiselection is enabled? It likely needs updating too.

Reply:
Are symbolic links to files handled correctly?

Reply:
In multiselection case the same issue exists where it doesn't allow user to multi select the Symbolic links. Code modification is required to handle multiselection case. And symbolic links to files are handled correctly, no issue with that.